core/src/kura.rs

```diff
@@ -255,6 +277,59 @@ impl Kura {
         Some(hash_array_guard[index])
     }
 
+    pub fn get_block_height_by_hash(&self, hash: &HashOf<VersionedCommittedBlock>) -> Option<u64> {
+        let hash_array_guard = self.block_hash_array.lock().expect("access hash array");
+
+        if *hash == Hash::zeroed().typed() {
+            None
+        } else {
+            match hash_array_guard.iter().position(|&x| x == *hash) {
+                None => None,
+                Some(index) => {
+                    Some(index as u64 + 1)
+                }
+            }
+        }
```

Comment:
I advocate for a monadic error handling model. 
```suggestion
        (*hash == Hash::zeroed().typed())
        .then_some(())
        .and(hash_array_guard.iter().position(|&x| x == *hash)
                .map(|index| index as u64 + 1))
```

Reply:
Given that the `hash` is unlikely to be `zero`, the order of logical checks should be reversed
```suggestion
       hash_array_guard.iter()
           .position(|&x| x == *hash)
           .map(|index| index as u64 + 1)
           .filter(|_| *hash == Hash::zeroed().typed())
```